signer/src/ecdsa.rs

```diff
@@ -6,10 +6,22 @@
 use codec::Encode;
 
 use crate::crypto::{seed_from_entropy, DeriveJunction, SecretUri};
+use core::str::FromStr;
+use derive_more::{Display, From};
 use hex::FromHex;
-use secp256k1::{ecdsa::RecoverableSignature, Message, SecretKey, SECP256K1};
+use secp256k1::{ecdsa::RecoverableSignature, All, Message, Secp256k1, SecretKey};
 use secrecy::ExposeSecret;
 
+struct GlobalSecp256K1Context;
+impl core::ops::Deref for GlobalSecp256K1Context {
+    type Target = Secp256k1<All>;
+
+    fn deref(&self) -> &Self::Target {
+        static ONCE: once_cell::sync::OnceCell<Secp256k1<All>> = once_cell::sync::OnceCell::new();
+        ONCE.get_or_init(|| Secp256k1::new())
+    }
```

Comment:
I'd copy what Substrate does here to be on the safe side, which is to do:

```rust
#[cfg(feature = "std")]
let context = SECP256K1;
#[cfg(not(feature = "std"))]
let context = Secp256k1::signing_only();
```

In the sign/from_seed fns to select which context to use ie https://github.com/paritytech/polkadot-sdk/blob/d54412ce8bef113c7a515b3a08851e7d77553df5/substrate/primitives/core/src/ecdsa.rs#L390 and https://github.com/paritytech/polkadot-sdk/blob/d54412ce8bef113c7a515b3a08851e7d77553df5/substrate/primitives/core/src/ecdsa.rs#L460

Since only two places this context is used anyway, I think that's fine :)